lib/3scale/backend/stats/keys/generator/application.rb

```diff
@@ -0,0 +1,58 @@
+module ThreeScale
+  module Backend
+    module Stats
+      module Keys
+        module Generator
+          class Application
+            def initialize(service_context, index_limits)
+              self.service_context = service_context
+              self.index_limits = index_limits
+              self.applications = service_context.applications || []
```

Comment:
This was made this way because the service_context is a private attribute accessor and self cannot be used when having a private attribute. For coherence should be getting the information from the attribute and not from the parameter.

I'll change all the classes to use @ directly in the constructors to set/get attributes of the class, and use the attribute getter/setter methods on the other methods of the class.

The idea to use getter/setter attributes is to be able to change the behaviour of them without having to update the @ references in the class code on the other methods